File: python/1045.py
x = input().split()
x = [float(x[0]), float(x[1]), float(x[2])]
x.sort()

a = x[2]
b = x[1]
c = x[0]

if a >= (b+c):
    print('NAO FORMA TRIANGULO')
else:
    if a * a == b * b + c * c:
        print('TRIANGULO RETANGULO')
    if a * a > b * b + c * c:
        print('TRIANGULO OBTUSANGULO')
    if a * a < b * b + c * c:
        print('TRIANGULO ACUTANGULO')
    if a == c and c == b:
        print('TRIANGULO EQUILATERO')
    if (a == c and b != a) or (a == b and a != c) or (b == c and b != a):
        print('TRIANGULO ISOSCELES')

File: lua/1045.lua
local entry = io.read()
local t = {}
for v in string.gmatch(entry, '[^%s]+') do
    table.insert(t, tonumber(v))
end

table.sort(t, function (a, b) return a > b end)

local a = t[1]
local b = t[2]
local c = t[3]

if a >= b + c then
    print('NAO FORMA TRIANGULO')
else
    if a^2 == b^2 + c^2 then
    print('TRIANGULO RETANGULO')
    end

    if a^2 > b^2 + c^2 then
        print('TRIANGULO OBTUSANGULO')
    end

    if a^2 < b^2 + c^2 then
        print('TRIANGULO ACUTANGULO')
    end

    if a == b and a == c then
        print('TRIANGULO EQUILATERO')
    elseif a == b or b == c or c == a then
        print('TRIANGULO ISOSCELES')
    end
end